modules/apps/layout/layout-reports-web/src/main/java/com/liferay/layout/reports/web/internal/display/context/LayoutReportsDisplayContext.java

```diff
@@ -211,9 +212,7 @@ private String _getConfigureGooglePageSpeedURL(
 				"mvcRenderCommandName",
 				"/configuration_admin/edit_configuration");
 			portletURL.setParameter(
-				"redirect",
-				_portal.getCurrentCompleteURL(
-					_portal.getHttpServletRequest(portletRequest)));
+				"redirect", _portal.getLayoutURL(_themeDisplay));
```

Comment:
That change makes totally sense, I need to use it in oder to calculate a valid canonical URL, very good catch :)

Reply:
I think we have a method called _getCompleteURL, that call this method and if yoiu find an exception returns the currentCompleteURL, what do you think about using it?